vertx-config-vault/src/main/java/io/vertx/config/vault/VaultConfigStore.java

```diff
@@ -0,0 +1,256 @@
+package io.vertx.config.vault;
+
+import io.vertx.config.spi.ConfigStore;
+import io.vertx.config.vault.client.*;
+import io.vertx.core.*;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonObject;
+
+import java.util.Objects;
+
+/**
+ * An implementation of {@link ConfigStore} for Vault (https://www.vaultproject.io/).
+ *
+ * @author <a href="http://escoffier.me">Clement Escoffier</a>
+ */
+public class VaultConfigStore implements ConfigStore {
+
+  private final SlimVaultClient client;
+  private final JsonObject config;
+  private final String path;
+  private final Vertx vertx;
+
+  /**
+   * When authenticated, whether or not the token is renewable.
+   */
+  private boolean renewable;
+
+  /**
+   * The time when the token must be renewed.
+   */
+  private long validity;
+
+  private Context context;
+
+  /**
+   * Creates an instance of {@link VaultConfigStore}.
+   *
+   * @param vertx  the vert.x instance
+   * @param config the configuration, used to create the {@link SlimVaultClient}.
+   */
+  public VaultConfigStore(Vertx vertx, JsonObject config) {
+    client = new SlimVaultClient(vertx, config);
+    this.config = config;
+    this.vertx = vertx;
+    this.path = Objects.requireNonNull(config.getString("path"), "The path of the secret must be set");
+  }
+
+  @Override
+  public void close(Handler<Void> completionHandler) {
+    client.close();
+    completionHandler.handle(null);
+  }
+
+  @Override
+  public void get(Handler<AsyncResult<Buffer>> completionHandler) {
+    if (context == null) {
```

Comment:
well, it seems I should add the same into `vertx-config-consul` module and keep `ConsulClient` as is...